sdk/src/test/java/com/finbourne/lusid/integration/HoldingsTests.java

```diff
@@ -0,0 +1,271 @@
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class HoldingsTests {
+
+    private TestDataUtiliites   testDataUtiliites;
+
+    private TransactionPortfoliosApi transactionPortfoliosApi;
+    private List<String> instrumentIds;
+
+    private final String SCOPE = "finbourne";
+
+    @Before
+    public void setUp() throws Exception {
+        File configJson = new TestConfigurationLoader().loadConfiguration("secrets.json");
+        ApiClient apiClient = new ApiClientBuilder(configJson).build();
+
+        this.transactionPortfoliosApi = new TransactionPortfoliosApi(apiClient);
+
+        this.testDataUtiliites = new TestDataUtiliites(this.transactionPortfoliosApi);
+
+        //  ensure instruments are created and exist in LUSID
+        this.instrumentIds = new InstrumentLoader().loadInstruments();
+        this.instrumentIds.sort(Comparator.naturalOrder());
+    }
+
+    @Test
+    public void get_holdings() throws ApiException {
+
+        final String    currency = "GBP";
+
+        OffsetDateTime  day1 =  OffsetDateTime.of(2018, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime  day2 =  OffsetDateTime.of(2018, 1, 5, 0, 0, 0, 0, ZoneOffset.UTC);
+        OffsetDateTime  day3 =  OffsetDateTime.of(2018, 1, 10, 0, 0, 0, 0, ZoneOffset.UTC);
+
+        String portfolioCode = this.testDataUtiliites.createTransactionPortfolio(SCOPE);
+
+        List<TransactionRequest>    requests = new ArrayList<>();
+
+        //  add starting cash
+        requests.add(testDataUtiliites.buildCashFundsInTransactionRequest(100000.0, currency, day1));
+
+        //  add initial transactions
+        requests.add(testDataUtiliites.buildTransactionRequest(this.instrumentIds.get(0), 100.0, 101.0, currency, day1, "Buy"));
+        requests.add(testDataUtiliites.buildTransactionRequest(this.instrumentIds.get(1), 100.0, 102.0, currency, day1, "Buy"));
+        requests.add(testDataUtiliites.buildTransactionRequest(this.instrumentIds.get(2), 100.0, 103.0, currency, day1, "Buy"));
+
+        //  add transactions on T+5, increasing the amount of instrument 1
+        requests.add(testDataUtiliites.buildTransactionRequest(this.instrumentIds.get(1), 100.0, 104.0, currency, day2, "Buy"));
+        requests.add(testDataUtiliites.buildTransactionRequest(this.instrumentIds.get(3), 100.0, 105.0, currency, day2, "Buy"));
+
+        //  add the trades
+        this.transactionPortfoliosApi.upsertTransactions(SCOPE, portfolioCode, requests);
+
+        //  get the holds on T+10
+        VersionedResourceListOfPortfolioHolding holdings = this.transactionPortfoliosApi.getHoldings(SCOPE, portfolioCode, false, day3,
+                null, null, null, null, null, null);
+
+        holdings.getValues().sort(Comparator.comparing(PortfolioHolding::getInstrumentUid));
+
+        assertThat(holdings.getCount(), is(equalTo(5)));
+
+        //  cash balance
+        assertThat(holdings.getValues().get(0).getInstrumentUid(), is(equalTo("CCY_" + currency)));
```

Comment:
This might be neater by putting holdings into a dictionary, then writing
`assertThat(holdings.get("CCY_" + currency), is(matchingHoldingType("B"))`

and

`assertThat(holdings.get("CCY_" + currency), is(matchingUnits("B"))`